main.py

```diff
@@ -490,7 +490,53 @@ def index(self):
             p.type = point.type
             points.append(p)
         context = {'points': points}
-        return self.render('viewhighlighted.html', **context)
+        return self.render('viewhighlighteddata.html', **context)
+
+    def is_visible(self):
+        return login.current_user.is_authenticated()
+
+class ViewHighlightedMarkersMap(BaseView):
```

Comment:
Doesn't this just show the same map as `index`, with the accident markers and the discussion markers? I don't see the highlight points being loaded here.
Also, we should avoid code duplication. If there's a lot of common code with `index`, please factor it out to a function.